### PR TITLE
Billing small fixes

### PR DESCRIPTION
**Pull request by milo-hyben**

### CODE REVIEW

File: web/src/pages/billing/components/BatchGrid.tsx [hunk 1]
```diff
@@ -279,7 +280,8 @@ const BatchCard: React.FC<{ item: AnalysisCostRecordBatch }> = ({ item }) => {
                     >
                         Cost by Sequencing Groups -{' '}
                         {item.seq_groups?.filter((s) => !!s.sequencing_group)?.length || 0}{' '}
-                        sequencing group(s)
+                        sequencing group(s) across{' '}
+                        {item.seq_groups?.filter((s) => !!s.stage)?.length || 0} stages
```
Comment (illusional) [inline web/src/pages/billing/components/BatchGrid.tsx:284]: This is the number of stage / sg combos right, my expectation would be the number of distinct stages across all sequencing groups, eg:

```
SG1 <stage1>
SG2 <stage1>
```
Should be 2 sequencing group(s) across 1 stage(s)

Comment (milo-hyben) [reply web/src/pages/billing/components/BatchGrid.tsx:284]: Well, just testing it and it is wrong it double count the stages per each seq group.

**Comment by milo-hyben** [comment]
Fixed the seq group counting and as well for the first 2 days of the month show last invoice month with all the budget details.

File: web/src/pages/billing/components/BatchGrid.tsx [hunk 2]
```diff
@@ -200,7 +200,13 @@ const AnalysisRunnerRecordCard: React.FC<{ data: AnalysisCostRecord }> = ({ data
...
                             Cost by Sequencing Groups -{' '}
                             {data.seq_groups?.filter((s) => !!s.sequencing_group)?.length || 0}{' '}
-                            sequencing group(s)
+                            sequencing group(s) across{' '}
+                            {new Set(
+                                data.seq_groups
+                                    ?.filter((s) => s.stage && s.stage.trim() !== '')
+                                    .map((s) => s.stage)
+                            ).size || 0}{' '}
+                            stages
```
Comment (illusional) [inline web/src/pages/billing/components/BatchGrid.tsx:209]:
```suggestion
                            stage(s)
```

File: web/src/pages/billing/components/BatchGrid.tsx [hunk 3]
```diff
@@ -279,7 +285,13 @@ const BatchCard: React.FC<{ item: AnalysisCostRecordBatch }> = ({ item }) => {
...
                         Cost by Sequencing Groups -{' '}
                         {item.seq_groups?.filter((s) => !!s.sequencing_group)?.length || 0}{' '}
-                        sequencing group(s)
+                        sequencing group(s) across{' '}
+                        {new Set(
+                            item.seq_groups
+                                ?.filter((s) => s.stage && s.stage.trim() !== '')
+                                .map((s) => s.stage)
+                        ).size || 0}{' '}
+                        stages
```
Comment (illusional) [inline web/src/pages/billing/components/BatchGrid.tsx:294]:
```suggestion
                        stage(s)
```

**Review by illusional — APPROVED** [review]

**Pull request merged**